### PR TITLE
Add and update tests

### PR DESCRIPTION
**Pull request by Smit1603**

### CODE REVIEW

**Comment by Smit1603** [comment]
@vgvassilev 
This PR has following changes made
- Added GetEnums test to EnumReflectionTest.cpp
- Removed extra blank space problem as visible earlier hence fixes #84
- As well as found a small error in findBuiltinType function hence resolved it
- Added some of the edge cases for some of the tests so that more code shall be covered

File: include/clang/Interpreter/CppInterOp.h [hunk 1]
```diff
@@ -173,6 +173,9 @@ namespace Cpp {
   /// Checks if the passed value is an enum type or not.
   bool IsEnumType(TCppType_t type);
 
+  /// Returns vector of enums present in given scope
+  std::vector<std::string> GetEnums(TCppScope_t scope);
```
Comment (vgvassilev) [inline include/clang/Interpreter/CppInterOp.h:177]: I am wondering where is the implementation of `GetEnums`?

Comment (Smit1603) [reply include/clang/Interpreter/CppInterOp.h:177]: Its already implemented in CppInterOp.cpp

Comment (vgvassilev) [inline include/clang/Interpreter/CppInterOp.h:177]:
```suggestion
  void GetEnums(TCppScope_t scope, std::vector<std::string>& Result);
```

We should change the signature in this way.

File: include/clang/Interpreter/CppInterOp.h [hunk 2]
```diff
@@ -173,6 +173,9 @@ namespace Cpp {
   /// Checks if the passed value is an enum type or not.
   bool IsEnumType(TCppType_t type);
 
+  /// Returns vector of enums present in given scope
```
Comment (vgvassilev) [inline include/clang/Interpreter/CppInterOp.h:176]: We should update the comment as the function does not return a vector anymore.

**Review by vgvassilev — APPROVED** [review]
LGTM!
